media-libs/libfdk_aac/libfdk_aac-0.1.5.recipe

```diff
@@ -0,0 +1,70 @@
+SUMMARY="Fraunhofer AAC codec library"
+DESCRIPTION='The Fraunhofer FDK AAC Codec Library for Android ("FDK AAC Codec") \
+	is software that implements the MPEG Advanced Audio Coding ("AAC") \
+	encoding and decoding scheme for digital audio.'
+HOMEPAGE="https://sourceforge.net/projects/opencore-amr"
+COPYRIGHT="1995-2012 Fraunhofer-Gesellschaft zur Förderung der angewandten Forschung e.V."
+LICENSE="Fraunhofer FDK AAC Codec"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/opencore-amr/fdk-aac/fdk-aac-$portVersion.tar.gz"
+CHECKSUM_SHA256="2164592a67b467e5b20fdcdaf5bd4c50685199067391c6fcad4fa5521c9b4dd7"
+SOURCE_DIR="fdk-aac-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="1.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libfdk_aac$secondaryArchSuffix = $portVersion
+	lib:libfdk_aac$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libfdk_aac${secondaryArchSuffix}_devel = $portVersion
+	devel:libfdk_aac$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	libfdk_aac$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:awk
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage libfdk_aac$secondaryArchSuffix \
+	$libDir/libfdk-aac.so.$libVersion
+
+BUILD()
+{
+	runConfigure ./configure --enable-shared
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	install -d -m 755 "$docDir"
+	install -t "$docDir" -m 644 ChangeLog
+
+	rm -rf "$libDir"/*.la
+
+	fixPkgconfig
+	prepareInstalledDevelLib libfdk-aac
+	packageEntries devel \
+		"$developDir" \
+		"$docDir/ChangeLog"
```

Comment:
In case we ship the 2 pdf files in the devel package, we need this tiny change:
```diff
 	packageEntries devel \
-		"$docDir/ChangeLog"
+		"$docDir"/ChangeLog \
+		"$docDir"/*.pdf
```


Shall we add this after INSTALL?
```

TEST()
{
# "check" target exists but does nothing
	make check
}
```

Reply:
No-op check is no better than not having it, so I would disagree with this. If one day someone actually add check to the library, then we can proceed adding `TEST()` for the port

Reply:
Well, there isn't really a consensus about this.
Some would argue that adding such a TEST() function with a comment stating that it does nothing makes it easier to find those other recipes which might do not have a TEST() function but have real tests.
Maybe we could/should add a TRUST_TEST variable with the following possible values :
* `yes` to indicate that a test suite exists and, in case the tests fail, this should trgger something, e.g. don't push the binary package to a public repo.
* `no` to indicate that a test suite exists but its result should be ignored. Some recipes work fine even if their tests fail.
* `fake` to indicate that the test suite does nothing.

Reply:
If the upstream decides to have no tests, it is their problem, not ours.

So, from our side we can consider we did everything needed. If the software turns out not to work and lands in the package repo, then it's upstream problem and they will have to deal with fixing the bugs in a hurry and the bad reputation.

So, empty test function (when there is nothing better provided by upstream) is enough, and there is no need to track the difference on our side.

Reply:
What do you call an empty test function? No TEST() function at all, or a TEST() function which calls something like "make check" which does nothing?

Reply:
one that calls a do-nothing operation (empty "make check", or "true", or "#no tests"). This way we clearly mark that we investigated the problem, and found nothing to test.

Reply:
OK, so I understand that you agree with me that having a TEST() function, even if is useless, is better than having no TEST() function at all.

This is where a `TRUST_TEST=fake` parameter makes sense. We would know that a test suite exists but does nothing. With such a parameter we would make everyone happy: those contributors who don't like empty TEST() functions, and those who like to know that a recipe has already been investigated about its tests.
We don't even need to change haikuporter. Adding a `TRUST_TEST=fake` (or anything more appropriate) cannot harm, as it will be ignored by haikuporter.

As usual, this is just a suggestion. If this is not an improvement, then let's forget about this.
We currently have some recipes with empty/fake tests, and some (most?) of these have a comment about that. But if we wanted to query the list of recipes in that case, there would be no easy method. With my suggestion it would be easy.
Needless to say, this TRUST_TEST would not need to be mandatory. It would perfectly make sense to use `TRUST_TEST=fake` only on those recipes with empty/fake tests, and leave all the other recipes unchanged, i.e. without any TRUST_TEST.

Reply:
This discussion is more about a META issue with haikuports than this PR. I suggest we move this conversation into an issue